querybook/webapp/lib/data-doc/datadoc-permission.ts

```diff
@@ -64,14 +69,21 @@ export function getViewerInfo(
     uid: number,
     editorsByUserId: Record<number, IDataDocEditor>,
     dataDoc: IDataDoc,
-    viewerIds: number[]
+    viewerIds: number[],
+    nonExplicitEditorPermissions: Record<number, IDataDocEditor>
 ): IViewerInfo {
-    const editor = uid in editorsByUserId ? editorsByUserId[uid] : null;
+    const editor =
+        uid in editorsByUserId
+            ? editorsByUserId[uid]
+            : uid in nonExplicitEditorPermissions
+            ? nonExplicitEditorPermissions[uid]
+            : null;
```

Comment:
not a fan of nested ternary operators, which is difficult to read and understand. could you rewrite it to make it more readable?